Emulators upgrades and refactors

AndersonTorres opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

tinyemu: 2018-09-23 -> 2019-12-21

##### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

c0bw3b commented:
Isn't it better to create a `pkgs/applications/emulators/` category?

The top-level `pkgs/misc` should contain _less_ things I think.
It's always possible to suggest a new tree under `applications` or `tools`.
Even using `pkgs/applications/misc` or `pkgs/tools/misc` seems less indiscriminate than the top-level misc.

AndersonTorres commented:
> Isn't it better to create a `pkgs/applications/emulators/` category?

I will do better then: move every those guys to `misc/emulation/` and "conflate" simulators, emulators and virtualizers in a same folder.

> 
> The top-level `pkgs/misc` should contain _less_ things I think. It's always possible to suggest a new tree under `applications` or `tools`. Even using `pkgs/applications/misc` or `pkgs/tools/misc` seems less indiscriminate than the top-level misc.

Indeed I think vim-plugins should live inside vim.

AndersonTorres committed:
Move misc/emulators to applications/emulators - part 1

Emulators form a class by themselves. So, they should be moved to applications/.

AndersonTorres committed:
Move misc/emulators to applications/emulators - part 2

Three small leftovers found via ripgrep.

AndersonTorres committed:
treewide: move some emulators to their correct place

bochs, dynamips, hercules, tiny8086 and tinyemu are not virtualizers.

AndersonTorres committed:
dynamips: refactor

AndersonTorres committed:
hercules: refactor

AndersonTorres committed:
tiny8086: refactor

AndersonTorres committed:
tinyemu: 2018-09-23 -> 2019-12-21
